core/io/json.cpp

```diff
@@ -588,10 +588,758 @@ void JSON::_bind_methods() {
+			Dictionary d;
+			PackedFloat64Array arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				values.push_back(arr[i]);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		case Variant::PACKED_STRING_ARRAY: {
+			Dictionary d;
+			PackedStringArray arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				values.push_back(arr[i]);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		case Variant::PACKED_VECTOR2_ARRAY: {
+			Dictionary d;
+			PackedVector2Array arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				Vector2 v = arr[i];
+				values.push_back(v.x);
+				values.push_back(v.y);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		case Variant::PACKED_VECTOR3_ARRAY: {
+			Dictionary d;
+			PackedVector3Array arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				Vector3 v = arr[i];
+				values.push_back(v.x);
+				values.push_back(v.y);
+				values.push_back(v.z);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		case Variant::PACKED_COLOR_ARRAY: {
+			Dictionary d;
+			PackedColorArray arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				Color v = arr[i];
+				values.push_back(v.r);
+				values.push_back(v.g);
+				values.push_back(v.b);
+				values.push_back(v.a);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		case Variant::PACKED_VECTOR4_ARRAY: {
+			Dictionary d;
+			PackedVector4Array arr = p_variant;
+			Array values;
+			for (int i = 0; i < arr.size(); i++) {
+				Vector4 v = arr[i];
+				values.push_back(v.x);
+				values.push_back(v.y);
+				values.push_back(v.z);
+				values.push_back(v.w);
+			}
+			d[VALUES] = values;
+			d[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+			return d;
+		} break;
+		default: {
+			ERR_PRINT(vformat("Unhandled conversion from native Variant type '%s' to JSON.", Variant::get_type_name(p_variant.get_type())));
+		} break;
+	}
+
+	Dictionary nil;
+	nil[GDTYPE] = Variant::get_type_name(p_variant.get_type());
+	return nil;
+}
+
+// TODO: fire 20240601 - Add p_allow_callable?
+Variant JSON::to_native(const Variant &p_json, bool p_allow_classes, bool p_allow_scripts) {
+	switch (p_json.get_type()) {
+		case Variant::BOOL: {
+			return p_json;
+		} break;
+		case Variant::INT: {
+			return p_json;
+		} break;
+		case Variant::FLOAT: {
+			return p_json;
+		} break;
+		case Variant::STRING: {
+			return p_json;
+		} break;
+		case Variant::STRING_NAME: {
+			return p_json;
+		} break;
+		case Variant::CALLABLE: {
+			// TODO: fire 20240602 - Implement handling of callable.
+			return p_json;
+		} break;
```

Comment:
I would prefer deciding on the TODOs now, instead of merging with something that will silently process Callable as a no-op. This kind of TODOs is the kind that never gets removed from the codebase until a cleanup is done, so best not add them in the first place, unless we are committed to implementing them.

I'm not sure Callables can/should be serialized.

Reply:
We decided earlier that callable is a no operation.

So is it ok to remove this todo? I'll do a scan for other todos.